### PR TITLE
Multi runtimes support

### PR DESCRIPTION
**Pull request by wuminzhe**

### CODE REVIEW

File: darwinia/src/darwinia.rs [hunk 1]
```diff
@@ -190,7 +199,8 @@ impl Darwinia {
 				}
 				let mut mmr_root: [u8; 32] = [0; 32];
 				mmr_root.copy_from_slice(&parent_mmr_root);
```
Comment (xiaoch05) [inline darwinia/src/darwinia.rs:201]: should we also delete these two lines?

Comment (wuminzhe) [reply darwinia/src/darwinia.rs:201]: yes

File: primitives/src/runtimes/mainnet.rs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+struct Log {
+	address: H160,
+	topics: Vec<H256>,
+	data: Vec<u8>,
+}
+
+/// Darwinia Runtime
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct MainnetRuntime;
```
Comment (HackFisher) [inline primitives/src/runtimes/mainnet.rs:50]: Since we have two mainnets Darwinia and Crab, I suggest to use Darwinia directly here.

**Review by HackFisher — CHANGES_REQUESTED** [review]

**Review by HackFisher — COMMENTED** [review]
@xiaoch05 Please help review and merge this pull request.

File: primitives/src/runtimes/mainnet.rs (hunk 2 above)
Comment (wuminzhe) [reply primitives/src/runtimes/mainnet.rs:50]: for now, I use mainnet, crab, pangolin as chain names.

The crate's name is `darwinia`. if we use `DarwiniaRuntime`, is it better to change the crate's name to a new one like `chain`?

Comment (HackFisher) [reply primitives/src/runtimes/mainnet.rs:50]: We should use the runtime spec name here, crab is not a darwinia chain.

Though the crab runtime is located at darwinia repo, we follow the way of polkadot repo, for the crate names, how about runtime-client-types, or runtime-types.

Non darwinia runtime could also locate here, e.g. polkadot or kusama or some other substrate runtime.

**Comment by xiaoch05** [comment]
Is there any other ways to set special Runtime such as `MainnetRuntime`?
There are too many positions to set this `MainnetRuntime `. And later for other runtimes, we must copy all of these codes many times.

**Comment by wuminzhe** [comment]
I did not make the code that uses 'Runtime' generic. You can do that for less codes.

File: src/api/darwinia_api.rs [hunk 3]
```diff
@@ -3,24 +3,25 @@ use darwinia::{
 	Darwinia, Darwinia2Ethereum, DarwiniaAccount, Ethereum2Darwinia, FromEthereumAccount,
 	ToEthereumAccount,
 };
+use primitives::runtimes::darwinia::DarwiniaRuntime;
```
[the remark this answers is not in the document]
Comment (wuminzhe) [reply src/api/darwinia_api.rs:6]: Great

**Comment by HackFisher** [comment]
In fact, the definition of runtime client types in bridger might be better to locate together with runtime, in darwinia-common/darwinia same repo.

Just like: https://github.com/darwinia-network/darwinia-common/pull/605/files?file-filters%5B%5D=.rs&file-filters%5B%5D=.toml#diff-e401ad5fc8fcd05972dc2243897b1c62a3b6fe5b430ce996dfa450098594820a

In this way, when bridger need to upgrade types, it only need to bump its dependency version.

**Pull request closed without merging**